File: timeConversion/timeConversion.ts
const timeConversion = (s: string):string /*lo que devuelte la funcion*/ => {
    const time : string[] = s.slice(0, 8).split(":");
    const minutes : number = parseInt(time[1], 10);
    const seconds:number  = parseInt(time[2], 10);
    let hours:number = parseInt(time[0], 10);
  
    const PM: string = s.slice(-2);
    if (PM == "PM" && hours != 12) hours += 12;
    else if (PM != "PM" && hours === 12) hours = 0;
  
    const formatHours: string = hours.toString().padStart(2, "0");
    const formatMinuts: string = minutes.toString().padStart(2, "0");
    const formatSeconds: string = seconds.toString().padStart(2, "0");
  
    return `${formatHours + ":" + formatMinuts + ":" + formatSeconds}`;
  };
  
  const string: string = "12:05:45PM";
  console.log(timeConversion(string));
  

File: timeConversion/TimeConversion.java

public class TimeConversion {

    public static String timeConversion(String s){
       String[] time = s.substring(0,8).split(":"); 
       int minutes = Integer.parseInt(time[1]);
       int seconds = Integer.parseInt(time[2]);
       int hours = Integer.parseInt(time[0]);
       

       String PM = s.substring(s.length() - 2);
       if(PM.equals("PM") & hours != 12){
        hours += 12;
       }
       else if (!PM.equals("PM") & hours == 12){
        hours = 0;
       };
       
        String formatHour = String.format("%02d",hours);
        String formatMinutes = String.format("%02d",minutes);
        String formatSeconds = String.format("%02d",seconds);

       return formatHour +":"+ formatMinutes+":"+ formatSeconds;
    }

    public static void main (String[] args) {
        String times = "07:05:15PM";
        System.out.println(timeConversion(times));
    }
}
